modules/frontend/pipeline/async_query_validator_middleware.go

```diff
@@ -45,6 +47,11 @@ func (c queryValidatorWare) validateTraceQLQuery(queryParams url.Values) error {
 		if err != nil {
 			return fmt.Errorf("invalid TraceQL query: %w", err)
 		}
+
+		// reject query if the query expression size exceeds the maximum allowed size
+		if len(traceQLQuery) > c.maxQuerySizeBytes {
```

Comment:
We could check the length of the string before parsing the query

Reply:
nice, this would allow us to skip parsing huge queries.